File: selection_insertion/selectionSort.cpp
#include<iostream>
#include<climits>
using namespace std;

int main(){
    int arr[]={5,4,2,1,3};
    int n=5;
    for(int ele : arr){
        cout<<ele<<" ";
    }
    cout<<endl;
    //insertion sort
    for(int i=0;i<n-1;i++){
        int min=INT_MAX;
        int minIndex=-1;
        for(int j=i;j<n;j++){
            if(arr[j]<min){
                min=arr[j];
                minIndex=j;
            }
        }
        swap(arr[i],arr[minIndex]);
    }
    for(int ele : arr){
        cout<<ele<<" ";
    }
    return 0;
}

File: java/selectionsort.java
public class selectionsort {
    public static void main(String[] args) {
        int [] arr = {64, 34, 25, 12, 22, 11, 90};
        int n = arr.length;
        for(int i=0;i<n-1;i++){
            int min=Integer.MAX_VALUE;
            int minIndex=i;
            for(int j=i;j<n;j++){
                if(arr[j]<min){
                    min=arr[j];
                    minIndex=j;
                }
            }
            int temp=arr[i];
            arr[i]=arr[minIndex];       
            arr[minIndex]=temp;
        }
        for(int i=0;i<n;i++){
            System.out.print(arr[i] + " ");
        }
    }
}
